sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIClientBuilder.java

```diff
@@ -290,13 +306,63 @@ private HttpPipeline createHttpPipeline() {
         return httpPipeline;
     }
 
+    private NonAzureOpenAIClientImpl buildInnerNonAzureOpenAIClient() {
+        HttpPipeline localPipeline = (pipeline != null) ? pipeline : createHttpPipelineNonAzureOpenAI();
+        NonAzureOpenAIClientImpl client =
+                new NonAzureOpenAIClientImpl(localPipeline, JacksonAdapter.createDefaultSerializerAdapter());
+        return client;
+    }
+
+    private HttpPipeline createHttpPipelineNonAzureOpenAI() {
```

Comment:
Nitpick: I would move this method to the `NonAzureOpenAIKeyCredentialPolicy` under a static method `CreateHttpPipeline()` (or any other better name you prefer), just to remove clutter from `OpenAIClientBuilder` if that makes sense? This method doesn't seem to be using any of the internal state of `OpenAIClientBuilder` so it seems just tangentially related.

Reply:
it is a builder-related method. We should keep it here. It probably could be more generic method after we have TypeSpec changes for Azure and NonAzure autho impl